src/pages/api/views/createNew.ts

```diff
@@ -0,0 +1,51 @@
+import { NextApiRequest, NextApiResponse } from 'next';
+
+import { createApiFetch } from 'utils/apiFetch';
+import { DATA_FIELD } from 'types/smartSearch';
+import postView from 'fetching/views/postView';
+import postViewColumn from 'fetching/views/postViewColumn';
+import { getBrowserLanguage, getMessages } from 'utils/locale';
+
+export default async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    const {
+        query: { orgId },
+        method,
+    } = req;
+
+    const lang = getBrowserLanguage(req);
+
+    const messages = await getMessages(lang, ['misc.views']);
+
+    // Return error if method other than POST
+    if (method !== 'POST') {
+        res.setHeader('Allow', ['POST']);
+        res.status(405).end(`Method ${method} Not Allowed`);
+    }
```

Comment:
I think this should be moved up (to bail earlier) and return after setting the status. Right now, the execution of this function will proceed to ty creating a view etc, right?